sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatAsyncClient.java

```diff
@@ -155,6 +164,62 @@ public void listChatThreadsNext(String nextLink, final Callback<Page<ChatThreadI
         this.serviceClient.listChatThreadsNext(nextLink, callback);
     }
 
+    /**
+     * Receive real-time messages and notifications.
+     */
+    public void startRealtimeNotifications() {
+        if (this.signalingClient == null) {
+            throw new Error("Signaling client not initialized");
+        }
+
+        if (this.isRealtimeNotificationsStarted) {
+            return;
+        }
+
+        this.isRealtimeNotificationsStarted = true;
+        this.signalingClient.start();
+    }
+
+    /**
+     * Stop receiving real-time messages and notifications.
+     */
+    public void stopRealtimeNotifications() {
+        if (this.signalingClient == null) {
+            throw new Error("Signaling client not initialized");
+        }
+
+        this.isRealtimeNotificationsStarted = false;
+        this.signalingClient.stop();
+    }
+
+    /**
+     * Listen to a chat event.
+     */
+    public void on(String chatEventId, RealTimeNotificationCallback listener) {
```

Comment:
Is this api for attaching a listener for real time notification? I wonder we can come-up with a more specialized name instead of the generic name `on`. (the same comment for `off` method as well).

Reply:
something like `[add|set]RealtimeNotificationCallback`?

Reply:
We explicitly call it on and off in other languages, so I will have to keep it.